### PR TITLE
ESP8266 Support

### PR DESCRIPTION
**Pull request by peternlewis**
Added support for NodeMCU/ESP8266.

### CODE REVIEW

File: drivers/arduino/RGB_LED_Matrix_Lib/src/RGBLEDMatrix.cpp
Comment (michaelkamprath) [inline drivers/arduino/RGB_LED_Matrix_Lib/src/RGBLEDMatrix.cpp:380]: Shouldn't this be a function call? That is:
```
	timer0_detachInterrupt();
```
Note the addition of the parenthesis.

**Review by michaelkamprath — CHANGES_REQUESTED** [review]

**Comment by peternlewis** [comment]
On 13 Oct 2017, at 00:00, Michael Kamprath <notifications@github.com> wrote:
> Shouldn't this be a function call? That is:
> 
> 	timer0_detachInterrupt();
> 
> Note the addition of the parenthesis.

Yes.

I expect it never happens, when would stopScanning() ever be called?

Thanks,
   Peter.

-- 
Keyboard Maestro <https://www.keyboardmaestro.com/> Editors' Choice Award winner.
Macros for your Mac <https://download.keyboardmaestro.com/>
Join us on the Keyboard Maestro forum <https://forum.keyboardmaestro.com/>

**Comment by michaelkamprath** [comment]
@peternlewis I would call `stopScanning(void)` if I need to suspend the use of the interrupt or even updating of the screen for whatever reason. Admittedly, looking at this, I see the original code had a bug in that the `stopScanning(void)` method needs to be scoped to the `RGBLEDMatrix`. Obvious I don't use it much myself :-/ 

I will merge and fix it myself when I fix my own bug.

**Comment by peternlewis** [comment]
OK, I'll leave it to you.

**Review by michaelkamprath — APPROVED** [review]

**Pull request merged**
